File: NativeAPI/Python/b2_create_bucket.py
import requests
import json

api_url = '' # Provided by b2_authorize_account
account_authorization_token = '' # Provided by b2_authorize_account
account_id = '' # Provided by b2_authorize_account
bucket_name = "any-name-you-pick"  # 63 char max: letters, digits, and hyphen -
bucket_type = "allPrivate"  # Either allPublic or allPrivate

response = requests.post(
    f"{api_url}/b2api/v4/b2_create_bucket",
    headers={"Authorization": account_authorization_token},
    json={
        "accountId": account_id,
        "bucketName": bucket_name,
        "bucketType": bucket_type
    }
)
response_data = response.json()
print(json.dumps(response_data, indent=2))

File: NativeAPI/Java/B2CreateBucket.java
package com.backblaze.examples;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class B2CreateBucket {
    public static void main(String[] args) {
        String apiUrl = ""; // Provided by b2_authorize_account
        String accountAuthorizationToken = "";  // Provided by b2_authorize_account
        String accountId = ""; // Provided by b2_authorize_account
        String bucketName = ""; // Name of the bucket you want to create
        String bucketType = ""; // Either "allPublic" or "allPrivate"

        String postParams = "{" +
                "\"accountId\":\"" + accountId + "\"," +
                "\"bucketName\":\"" + bucketName + "\"," +
                "\"bucketType\":\"" + bucketType + "\"" +
                "}";

        try (HttpClient client = HttpClient.newHttpClient()) {
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(apiUrl + "/b2api/v4/b2_create_bucket"))
                    .header("Authorization", accountAuthorizationToken)
                    .POST(HttpRequest.BodyPublishers.ofString(postParams))
                    .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            String jsonResponse = response.body();
            System.out.println(jsonResponse);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
